source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -305,8 +305,18 @@ ShaderPtr GlslShaderGenerator::generate(const string& name, ElementPtr element,
 
 void GlslShaderGenerator::emitVertexStage(const ShaderGraph& graph, GenContext& context, ShaderStage& stage) const
 {
-    // Add version directive
+    HwResourceBindingContextPtr resourceBindingCtx =
+        context.getUserData<HwResourceBindingContext>(HW::USER_DATA_BINDING_CONTEXT);
+
+    // Add directives
     emitLine("#version " + getVersion(), stage, false);
+    if (resourceBindingCtx)
+    {
+        for (auto& extension : resourceBindingCtx->requiredExtensions())
```

Comment:
Would it be better to add in an emitExtenions() method on the context to hide this and avoid repeating the code?

Reply:
I wanted to make HwResourceBindingContextPtr binding context, only a maintainer of the extensions, but actual shader gen is done by the generator. I can move to HwResourceBindingContextPtr as `::emitDirectives()`